awx/main/tasks.py

```diff
@@ -494,29 +495,22 @@ def execution_node_health_check(node):
 
 def inspect_execution_nodes(instance_list):
     with advisory_lock('inspect_execution_nodes_lock', wait=False):
-        node_lookup = {}
-        for inst in instance_list:
-            if inst.node_type == 'execution':
-                node_lookup[inst.hostname] = inst
+        # {inst.hostname: inst for inst in instance_list}
+        node_lookup = Instance.objects.in_bulk(field_name='hostname')
 
         ctl = get_receptor_ctl()
-        connections = ctl.simple_command('status')['Advertisements']
+        mesh_status = ctl.simple_command('status')
+
         nowtime = now()
-        for ad in connections:
+        workers = mesh_status['Advertisements']
+        for ad in workers:
             hostname = ad['NodeID']
-            commands = ad.get('WorkCommands') or []
-            worktypes = []
-            for c in commands:
-                worktypes.append(c["WorkType"])
-            if 'ansible-runner' not in worktypes:
+            if not any(cmd['WorkType'] == 'ansible-runner' for cmd in ad['WorkCommands']):
```

Comment:
I'd rather not risk `KeyError` exceptions here. You could get assurances from the receptor team that "WorkCommands" will always be there, or you could just change it to a `.get()` with a fallback (the cheap solution)

Reply:
As it turns out, it is sometimes `None` (when it is a hop node), which I entirely forgot.  I've added a fallback of `or []`.